src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs

```diff
@@ -881,9 +883,10 @@ internal string DisplayName
                             // if DefaultThreadCurrentUICulture was set
                             CultureInfo ci;
 
+                            // TODO-NULLABLE: DefaultThreadCurrentUICulture checked for null
                             if (CultureInfo.DefaultThreadCurrentUICulture != null &&
                                 ((ci = GetUserDefaultCulture()) != null) &&
-                                !CultureInfo.DefaultThreadCurrentUICulture.Name.Equals(ci.Name))
+                                !CultureInfo.DefaultThreadCurrentUICulture!.Name.Equals(ci.Name))
```

Comment:
In effectively the same code earlier, you also had `ci!.Name` but here you just have `ci.Name`... was the compiler actually warning about that previous usage and not about this one?  Or can the `!` from the previous usage be removed?

Reply:
from what I've noticed once you `!` something the compiler doesn't complain about it in the same context

Reply:
Actually `!` is not sticky - it only changes the top-level nullability of the expression where it is applied. But when a variable is dereferenced, we assume the variable is not null after that point since an exception would be thrown otherwise.

For instance, with `void F(object arg)` and `object? x, y;`, there is only one warning in the following:
```C#
F(x!);
F(x); // warning
y!.ToString();
y.ToString();
```

---
In reply to: [270580597](https://github.com/dotnet/coreclr/pull/23526#discussion_r270580597) [](ancestors = 270580597)

Reply:
I'm not sure where before at this point. Either way - this particular warning doesn't show up anymore so I'm removing `!` here (pending)